src/main/java/ch/njol/skript/classes/data/BukkitClasses.java

```diff
@@ -1719,6 +1720,41 @@ public String getVariableNamePattern() {
 					})
 					.serializer(new EnumSerializer<>(Cat.Type.class)));
 		}
+		if (Skript.classExists("org.bukkit.enchantments.EnchantmentOffer")) {
+			Classes.registerClass(new ClassInfo<>(EnchantmentOffer.class, "enchantmentoffer")
+				.user("enchant[ment][ ]offers?")
+				.name("Enchantment Offer")
+				.description("The enchantmentoffer in an enchant prepare event.")
+				.examples("cost of enchantmentoffer")
+				.since("INSERT VERSION")
+				.parser(new Parser<EnchantmentOffer>() {
+					@Override
+					@Nullable
+					public EnchantmentOffer parse(String input, ParseContext context) {
+						return null;
+					}
+
+					@Override
+					public boolean canParse(ParseContext context) {
+						return false;
+					}
+
+					@Override
+					public String toString(EnchantmentOffer eo, int flags) {
+						return EnchantmentType.toString(eo.getEnchantment(), flags) + " " + eo.getEnchantmentLevel();
+					}
+	
+					@Override
+					public String toVariableNameString(EnchantmentOffer eo) {
+						return toString(eo, 0);
```

Comment:
This should be `"offer:" + eo.getEnchantment().getKey().getKey() + "=" + eo.getEnchantmentLevel()` or something similar, and the varible pattern may match that as well.